strongbox-web-core/src/main/java/org/carlspring/strongbox/services/DirectoryListingServiceImpl.java

```diff
@@ -0,0 +1,181 @@
+package org.carlspring.strongbox.services;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.sql.Date;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang.StringUtils;
+import org.carlspring.strongbox.configuration.ConfigurationManager;
+import org.carlspring.strongbox.domain.DirectoryListing;
+import org.carlspring.strongbox.domain.FileContent;
+import org.carlspring.strongbox.providers.io.RepositoryFiles;
+import org.carlspring.strongbox.providers.io.RepositoryPath;
+import org.carlspring.strongbox.storage.Storage;
+import org.carlspring.strongbox.storage.repository.Repository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DirectoryListingServiceImpl implements DirectoryListingService
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(DirectoryListingService.class);
+
+    @Inject
+    private ArtifactResolutionService artifactResolutionService;
+    
+    @Inject
+    private ConfigurationManager configurationManager;
+    
+    @Override
+    public DirectoryListing fromStorages(Map<String, Storage> storages) throws IOException
+    {
+        DirectoryListing directoryListing = new DirectoryListing();
+
+        for (Storage storage : storages.values())
+        {
+            FileContent fileContent = new FileContent(storage.getId());
+            directoryListing.getDirectories().add(fileContent);
+            
+            fileContent.setStorageId(storage.getId());
+            fileContent.setUrl(calculateDirectoryUrl(fileContent));
+        }
+
+        return directoryListing;
+    }
+
+    @Override
+    public DirectoryListing fromRepositories(Map<String, Repository> repositories) throws IOException
+    {
+        DirectoryListing directoryListing = new DirectoryListing();
+
+        for (Repository repository : repositories.values())
+        {
+            FileContent fileContent = new FileContent(repository.getId());
+            directoryListing.getDirectories().add(fileContent);
+            
+            fileContent.setStorageId(repository.getStorage().getId());
+            fileContent.setRepositoryId(repository.getId());
+            
+            fileContent.setUrl(calculateDirectoryUrl(fileContent));
+        }
+
+        return directoryListing;
+    }
+
+    @Override
+    public DirectoryListing fromPath(RepositoryPath path)
```

Comment:
I believe we should also preserve the previous implementation here `public DirectoryListing fromPath(Path rootPath, Path path)`. The path you might want to generate `DirectoryListing` for might actually be some FS path on the server outside the context of `RepositoryPath`, `Storage`, and `Repository` (i.e. like the browsing the logs or something else which might need this in the future).

Reply:
`RepositoryPath`  have this under the hood